### DISCUSSION TITLE
For specific mastodon instance, couldn't authorise user.getting "We're sorry, but something went wrong on our end."

### DISCUSSION THREAD

**Discussion by sudhagar-mr-17051** [mastodon/mastodon#25616] [Q&A]
While authorising user on mastodon in mastodon.world instance, getting "We're sorry, but something went wrong on our end." error page. Before it was working fine. provided the details below:

On removing any fields in state param, the auth page displays. Is state param characters counted? if so, where the count limit of state param is mentioned? if this was not the issue, kindly let us know about it. 

exact url: 

https://mastodon.world/oauth/authorize?scope=read+write+push&response_type=code&force_login=true&redirect_uri=*******************&client_id=5nuDTYsvOb9Lhn1h6SCEZ8dhkPvgC0rUUqHxFVW6DIA&state=%7B%22isorg%22%3Atrue%2C%22servername%22%3A%22mastodon.world%22%2C%22currentportal%22%3A%22sudhagarmr%22%2C%22onBoard%22%3Atrue%2C%22from%22%3A%22mastodon%22%2C%22prcode%22%3A%22ZR%22%2C%22key%22%3A%22socialenear3d4363f16510aeb6cd8707aae3d5a8f8fe3111395f82c63b324aa98de8b26c599c08c231391da70c9a3590134c5800ce%22%2C%22brand_id%22%3A150043000000019021%7D

[NOTE: have masked redirect_uri ]

**Comment by andypiper** [comment]
You might need to contact the instance owner in this case, possibly there is something up with network routing or the setup on their side. You say "before it was working fine", when did this change, and could it have been related to a change in config on either side?

**Reply by sudhagar-mr-17051** [reply]
We haven't changed anything from our end in OAUTH related packages. We also tried to contact mastodon.world admins but they aren't responding to any private messages on their instance. Is there any way to reach out instance owner?

**Reply by ruudschilders** [reply]
I will certainly look into this, I received your DM yesterday when I was busy with other stuff. Sorry for the late response.

**Comment by ruudschilders** [comment]
When did it last work?

**Reply by sudhagar-mr-17051** [reply]
We were in development stage while we checked last time(It was working fine), it was six weeks before and now while starting testing, we are getting this error, but it was working fine for other instances.

**Reply by ruudschilders** [reply]
We upgraded Mastodon to `nightly-2023-05-17` 6 weeks ago..
I'll see if there are any known issues with that version

**Reply by sudhagar-mr-17051** [reply]
Kindly let us know if that is the case. Thanks.

**Reply by renchap** [reply]
@sudhagar-msr are you able to reproduce this on mastodon.social or mastodon.online by any chance?

**Reply by sudhagar-mr-17051** [reply]
It is working fine in mastodon.social and mastodon.online,

But not works in mastodon.world and mastodonapp.uk

**Reply by sudhagar-mr-17051** [reply]
Hi, This message is to follow up on the issue raised above, is there any update regarding the same? If any further information needed kindly ask me

**Reply by renchap** [reply]
This may be caused by mastodon.world using an older release that the one we run on mastodon.social.

I dont remember particular changes involving this in the last 6 weeks, maybe @ClearlyClaire does?

**Reply by sudhagar-mr-17051** [reply]
The case was not with the versions, I think so, while analysing, found few things, which may help you resolve the issue.

mas.to and mstdn.social both are in versions 4.1.2

on making auth call to mstdn.social
url: https://mstdn.social/oauth/authorize?scope=read+write+push&response_type=code&force_login=true&redirect_uri=https%3A%2F%2Fsocialtest1.localzoho.com%2Fsocialcallback.do%3Ffrom%3Dmastodon&client_id=KM6HxhL-i-fyABMtMxAYRNaYJTy6tzVUjNAJrxr6rdQ&state=%7B%22isorg%22%3Atrue%2C%22servername%22%3A%22mstdn.social%22%2C%22currentportal%22%3A%22sudhagarmr%22%2C%22onBoard%22%3Atrue%2C%22from%22%3A%22mastodon%22%2C%22prcode%22%3A%22ZR%22%2C%22key%22%3A%22socialenearc5edc4a7818538952d7f4562194356844112c50b31b87bf20300743f219a5b7491448e1f271bbedf19cc27dcb617afb5%22%2C%22brand_id%22%3A150043000000019021%7D

getting "We're sorry, but something went wrong on our end." error page


on making call to mas.to

url: https://mas.to/oauth/authorize?scope=read+write+push&response_type=code&force_login=true&redirect_uri=https%3A%2F%2Fsocialtest1.localzoho.com%2Fsocialcallback.do%3Ffrom%3Dmastodon&client_id=b5EO2c-FaHMM2GdOll3Zq7xApxtJBbscUywqr5vdM4I&state=%7B%22isorg%22%3Atrue%2C%22servername%22%3A%22mas.to%22%2C%22currentportal%22%3A%22sudhagarmr%22%2C%22onBoard%22%3Atrue%2C%22from%22%3A%22mastodon%22%2C%22prcode%22%3A%22ZR%22%2C%22key%22%3A%22socialenear84acdb5de23e0c71e2373f06c886f8ad4da8bd5539922bf649f7b4edd5f851877b40bf78735fb42dcb17182dbcf853b6%22%2C%22brand_id%22%3A150043000000019021%7D

getting proper auth page and could able to authenticate user.

[NOTE: ON removing state param, it is working fine in all instances]

is state param length matters? if so does it vary for each instances? can you confirm all these ?

**Reply by sudhagar-mr-17051** [reply]
Hi any update on this issue will greatly help us.

**Reply by renchap** [reply]
Some servers might have limits on state param length, but if it works on some servers then it most probably does not come from Mastodon's code.

**Reply by ruudschilders** [reply]
We're upgrading mastodon.world this evening, please check after that if that helped.

**Reply by sudhagar-mr-17051** [reply]
Sure, will check, Thanks for the update from both of you!
